server/swap/swap.go

```diff
@@ -0,0 +1,1171 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package swap
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/decred/dcrdex/server/account"
+	"github.com/decred/dcrdex/server/asset"
+	"github.com/decred/dcrdex/server/comms"
+	"github.com/decred/dcrdex/server/comms/rpc"
+	"github.com/decred/dcrdex/server/matcher"
+	"github.com/decred/dcrdex/server/order"
+)
+
+const (
+	// These constants are used for a chainWaiter to inform the caller whether
+	// the waiter should be run again.
+	tryAgain     = false
+	dontTryAgain = true
+)
+
+var (
+	// The chainWaiters will query for transaction data every recheckInterval.
+	recheckInterval = time.Second * 5
+	zeroTime        = time.Time{}
+	// txWaitExpiration is the longest the Swapper will wait for a chainWaiter.
+	// This could be thought of as the maximum allowabloe backend latency.
+	txWaitExpiration = time.Minute
```

Comment:
This is not a DEX variable in the spec. I'm not sure if it should be or not.

Reply:
Note sure either, but it feels like an operator or server mesh config consensus decision.

allowabloe -> allowable